yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/DocumentDetailsAttributeValue.java

```diff
@@ -44,4 +47,71 @@ public String getIssuingAuthority() {
         return authority;
     }
 
+    @Override
+    public String toString() {
+        String basePattern = "%s %s %s";
+        String result = String.format(basePattern, type, issuingCountry, number);
+
+        if (expirationDate != null) {
+            result = result + " " + expirationDate;
+        }
+
+        if (!isNullOrEmpty(authority)) {
+            String separator = expirationDate == null ? " - " : " ";
+            result = result + separator + authority;
+        }
+
+        return result;
+    }
+
+    public static DocumentDetailsValueBuilder builder() {
+        return new DocumentDetailsValueBuilder();
+    }
+
+    public static class DocumentDetailsValueBuilder {
+
+        private String type;
+        private String issuingCountry;
+        private String number;
+        private Date expirationDate;
+        private String authority;
+
+        private DocumentDetailsValueBuilder() {
+        }
+
+        public DocumentDetailsValueBuilder withType(String type) {
```

Comment:
Yes, kind of.  It's a builder.  The _DocumentDetailsAttributeValue_ has a private constructor - to create an instance of it, you have to use its _DocumentDetailsValueBuilder_.
Its really just about code style - builders allow an elegant, readable way to create an instance of an object that may require several values:
```java
DocumentDetailsAttributeValue result = DocumentDetailsAttributeValue.builder()
                .withType(SOME_TYPE)
                .withIssuingCountry(SOME_COUNTRY)
                .withNumber(SOME_NUMBER)
                .build();
```

In this case we are allowing creation of a DocumenentDetailsAttributeValue so that it can be set on the sandbox.

We will allow creation of value objects, but we will not allow them to be set on the profile.  The ActivityDetails/HumanProfile/ApplicationProfile returned by the SDK must be immutable - we don't people to change the attributes that are on it.